ensime_shared/protocol.py

```diff
@@ -197,54 +193,52 @@ def handle_string_response(self, call_id, payload):
           2. `DebugToStringReq`
           3. `FormatOneSourceReq`
         """
-        self.log(str(payload))
+        self.log.debug('handle_string_response: in %s', payload)
         self.handle_doc_uri(call_id, payload)
 
     def handle_doc_uri(self, call_id, payload):
         """Handler for responses of Doc URIs."""
-        if not self.en_format_source_id:
-            self.log("handle_string_response: received doc path")
+        if not self.en_format_source_id:  # TODO: what does format source have to do with this?
+            self.log.debug('handle_doc_uri: received doc path')
             port = self.ensime.http_port()
 
             url = payload["text"]
 
             if not url.startswith("http"):
                 url = gconfig["localhost"].format(port, payload["text"])
 
-            browse_enabled = self.call_options[call_id].get("browse")
-
-            if browse_enabled:
-                log_msg = "handle_string_response: browsing doc path {}"
-                self.log(log_msg.format(url))
+            if self.call_options[call_id].get('browse'):
+                self.log.debug('handle_doc_uri: browsing doc path %s', url)
                 try:
                     if webbrowser.open(url):
-                        self.log("opened {}".format(url))
-                except webbrowser.Error as e:
-                    log_msg = "handle_string_response: webbrowser error: {}"
-                    self.log(log_msg.format(e))
-                    self.raw_message(feedback["manual_doc"].format(url))
+                        self.log.info('opened %s', url)
+                except webbrowser.Error:
+                    self.log.exception('handle_doc_uri: webbrowser error')
+                    self.editor.raw_message(feedback["manual_doc"].format(url))
 
             del self.call_options[call_id]
         else:
+            # TODO: Get rid of this self.vim usage, but I don't understand
+            # when/if we ever get here...
             self.vim.current.buffer[:] = \
                 [line.encode('utf-8') for line in payload["text"].split("\n")]
             self.en_format_source_id = None
```

Comment:
If anyone knows what this `else` branch is supposed to do, I'd appreciate being enlightened, because this is the only other remaining `self.vim` aside from `disable_plugin`.

`:EnDocBrowse` works on this branch, `:EnDocUri` is broken on master AFAICT and I've never really known what it's supposed to do/what the point of it is anyway. `:EnDocUri` doesn't hit this code path either, it fails in the `if` branch. No idea what `self.en_format_source_id` is supposed to have to do with this.